vhost_rs/src/vhost_user/message.rs

```diff
@@ -575,10 +575,9 @@ impl VhostUserMsgValidator for VhostUserConfig {
     fn is_valid(&self) -> bool {
         if (self.flags & !VhostUserConfigFlags::all().bits()) != 0 {
             return false;
-        } else if self.offset < VHOST_USER_CONFIG_OFFSET
-            || self.offset >= VHOST_USER_CONFIG_SIZE
+        } else if self.offset >= VHOST_USER_CONFIG_SIZE
```

Comment:
From your commit message, you explain this is specific to `virtio-mmio`. Could you detail what is different in this case between `virtio-mmio` and `virtio-pci`, and does it break `virtio-pci`?

Second thing, this crate is going to be used by VMM that supports `virtio-mmio` only, hence we need it to work for both transport layers. Rust-vmm will never accept this crate if it's exclusively `virtio-pci` oriented.

Reply:
@sboeuf ,  as for offset definition in virtio-mmio, please ref **4.2.2 MMIO Device Register Layout** in virtio spec 1.1

I also list the key info below:
virtio-mmio:
Configuration space
Device-specific configuration space starts at the **offset 0x100** and is accessed with byte alignment. Its meaning and size depend on the device and the driver

By the way, i talked with Liujiang, and we agreed this changes and will add offset in virtio-mmio side, not vhost user protocol.

Liujiang will release new vhost_rs release and one guy from redhat has submitted one patch for this, please ref https://github.com/jiangliu/vhost/commit/b5db8ec07d4914cca708a64af96e361a66f8a47a

Reply:
Yes that's what I was expecting. The change has to happen on the transport layer implementation, it should be independent from vhost-user.